keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py

```diff
@@ -0,0 +1,221 @@
+class DeepLabV3Plus(Task):
+    """A Keras model implementing the DeepLabV3+ architecture for semantic
+    segmentation.
+
+    References:
+        - [Encoder-Decoder with Atrous Separable Convolution for Semantic Image Segmentation](https://arxiv.org/abs/1802.02611)
+        (ECCV 2018)
+        - [Rethinking Atrous Convolution for Semantic Image Segmentation](https://arxiv.org/abs/1706.05587)
+        (CVPR 2017)
+
+    Args:
+        num_classes: int, the number of classes for the detection model. Note
+            that the `num_classes` doesn't contain the background class, and the
+            classes from the data should be represented by integers with range
+            [0, `num_classes`).
+        backbone: `keras.Model`. The backbone network for the model that is
+            used as a feature extractor for the DeepLabV3 Encoder. Should be a
+            `keras_cv.models.backbones.backbone.Backbone`. A somewhat sensible
+            backbone to use in many cases is the:
+            `keras_cv.models.ResNet50V2Backbone.from_preset("resnet50_v2_imagenet")`.
+        projection_filters: int, number of filters in the
+            convolution layer projecting low-level features from the `backbone`.
+            The default value is set to `48`, as per the
+            [TensorFlow implementation of DeepLab](https://github.com/tensorflow/models/blob/master/research/deeplab/model.py#L676).
+        spatial_pyramid_pooling: (Optional) a `keras.layers.Layer`. Also known
+            as Atrous Spatial Pyramid Pooling (ASPP). Performs spatial pooling
+            on different spatial levels in the pyramid, with dilation. If
+            provided, the feature map from the backbone is passed to it inside
+            the DeepLabV3 Encoder, otherwise
+            `keras_cv.layers.spatial_pyramid.SpatialPyramidPooling` is used.
+        segmentation_head: (Optional) a `keras.layers.Layer`. If provided, the
+            outputs of the DeepLabV3 encoder is passed to this layer and it
+            should predict the segmentation mask based on feature from backbone
+            and feature from decoder, otherwise a similar architecture is used.
+        dropout (Optional) float, The dropout rate applied to the outputs of the
+            decoder in the `segmentation_head`.
+
+    Examples:
+    ```python
+    import tensorflow as tf
+    import keras_cv
+
+    images = tf.ones(shape=(1, 96, 96, 3))
+    labels = tf.zeros(shape=(1, 96, 96, 1))
+    backbone = keras_cv.models.ResNet50V2Backbone(input_shape=[96, 96, 3])
+    model = keras_cv.models.segmentation.DeepLabV3Plus(
+        num_classes=1, backbone=backbone,
+    )
+
+    # Evaluate model
+    model(images)
+
+    # Train model
+    model.compile(
+        optimizer="adam",
+        loss=keras.losses.BinaryCrossentropy(from_logits=False),
+        metrics=["accuracy"],
+    )
+    model.fit(images, labels, epochs=3)
+    ```
+    """
+
+    def __init__(
+        self,
+        num_classes,
```

Comment:
Let's make `backbone` the first parameter (this is required for proper compatability with the `task` model, and it's our preferred API structure)